use of MY_ESP8266_SSID and MY_ESP8266_PASSWORD deprecated

Compiling the template .ino file configured as an ESP8266 gateway warns that the two #defines for specifying the SSID and Password for the WiFi network are deprecated and compilation fails.

#define MY_ESP8266_SSID "myssid"
#define MY_ESP8266_PASSWORD "mypassword"

 #warning MY_ESP8266_SSID is deprecated, use MY_WIFI_SSID instead!

  ^

C:\Users\Mitch.MAGNOLIAMANOR\Documents\Arduino\libraries\MySensors/core/MyGatewayTransportMQTTClient.cpp:33:2: warning: #warning MY_ESP8266_PASSWORD is deprecated, use MY_WIFI_PASSWORD instead! [-Wcpp]

 #warning MY_ESP8266_PASSWORD is deprecated, use MY_WIFI_PASSWORD instead!

--- a/MySensors_NodeManager.h
+++ b/MySensors_NodeManager.h
@@ -87,8 +87,8 @@ NodeManager nodeManager;
 #include "nodemanager/Sensor.cpp"
 // Child class
 #include "nodemanager/Child.cpp"
-// Timer class
-#include "nodemanager/Timer.cpp"
+// InternalTimer class
+#include "nodemanager/InternalTimer.cpp"
 
 #if NODEMANAGER_OTA_CONFIGURATION == ON
 // include SensorConfiguration if needed

--- a/README.md
+++ b/README.md
@@ -275,7 +275,7 @@ You can interact with each class provided by NodeManager through a set of API fu
 	void registerSensor(Sensor* sensor);
 #if NODEMANAGER_SLEEP == ON
 	// register a timer
-	void registerTimer(Timer* timer);
+	void registerTimer(InternalTimer* timer);
 #endif
 	// return the next-available child id
 	uint8_t getAvailableChildId(uint8_t child_id = 0);

--- a/examples/Template/Template.ino
+++ b/examples/Template/Template.ino
@@ -121,8 +121,8 @@ NodeManager. Just uncomment the settings you need and the sensors you want to ad
 
 // ESP8266 gateway settings
 //#define MY_GATEWAY_ESP8266
-//#define MY_ESP8266_SSID ""
-//#define MY_ESP8266_PASSWORD ""
+//#define MY_WIFI_SSID ""
+//#define MY_WIFI_PASSWORD ""
 
 // Gateway networking settings
 //#define MY_IP_ADDRESS 192,168,178,87

--- a/keywords.txt
+++ b/keywords.txt
@@ -23,7 +23,7 @@ Child	KEYWORD1
 ConfigurationRequest	KEYWORD1
 List	KEYWORD1
 PowerManager	KEYWORD1
-Timer	KEYWORD1
+InternalTimer	KEYWORD1
 
 # NodeManager functions
 

--- a/nodemanager/Child.h
+++ b/nodemanager/Child.h
@@ -24,7 +24,7 @@ Child: data structure for representing a Child of a Sensor
 */
 
 class Sensor;
-#include "Timer.h"
+#include "InternalTimer.h"
 
 // how many slots to leave for the user before starting using them for Child
 #define EEPROM_CHILD_OFFSET	10
@@ -125,7 +125,7 @@ protected:
 #if NODEMANAGER_CONDITIONAL_REPORT == ON
 	double _last_value = 0;
 	const char* _last_value_string = "";
-	Timer* _force_update_timer = new Timer();
+	InternalTimer* _force_update_timer = new InternalTimer();
 	float _min_threshold = -FLT_MAX;
 	float _max_threshold = FLT_MAX;
 	float _value_delta = 0;

--- a/nodemanager/Constants.h
+++ b/nodemanager/Constants.h
@@ -105,17 +105,11 @@ Chip type
 #endif
 
 // define interrupt pins
-
-#if defined(CHIP_STM32)
-#define INTERRUPT_PIN_1 PB8
-#define INTERRUPT_PIN_2 2
-#else
 #ifndef INTERRUPT_PIN_1
 #define INTERRUPT_PIN_1 3
 #endif
 #ifndef INTERRUPT_PIN_2
 #define INTERRUPT_PIN_2 2
-#endif
 #endif
 
 // define eeprom addresses

--- a/nodemanager/Constants.h
+++ b/nodemanager/Constants.h
@@ -100,7 +100,10 @@ Chip type
 #if defined (MYBOARDNRF5) || defined(NRF51) || defined(NRF52)
 #define CHIP_NRF5
 #endif
-#if !defined(CHIP_ESP8266) && !defined(CHIP_ESP32) && !defined(CHIP_STM32) && !defined(CHIP_NRF5)
+#if defined (SAMD_SERIES)
+#define CHIP_SAMD
+#endif
+#if !defined(CHIP_ESP8266) && !defined(CHIP_ESP32) && !defined(CHIP_STM32) && !defined(CHIP_NRF5) && !defined(CHIP_SAMD)
 #define CHIP_AVR
 #endif
 

--- a/nodemanager/InternalTimer.cpp
+++ b/nodemanager/InternalTimer.cpp
@@ -18,37 +18,37 @@
 */
 
 /******************************************
-Timer: helper class to keep track of the elapsed time
+InternalTimer: helper class to keep track of the elapsed time
 */
 
-#include "Timer.h"
+#include "InternalTimer.h"
 
-Timer::Timer() {
+InternalTimer::InternalTimer() {
 	nodeManager.registerTimer(this);
 }
 
 // set the timer mode
-void Timer::setMode(timer_mode mode) {
+void InternalTimer::setMode(timer_mode mode) {
 	_mode = mode;
 }
 
 // get the timer mode
-timer_mode Timer::getMode() {
+timer_mode InternalTimer::getMode() {
 	return _mode;
 }
 
 // set the timer value in seconds
-void Timer::setValue(unsigned long value) {
+void InternalTimer::setValue(unsigned long value) {
 	_value = value;
 }
 
 // get the timer value in seconds
-unsigned long Timer::getValue() {
+unsigned long InternalTimer::getValue() {
 	return _value;
 }
 
 // start/restart the timer
-void Timer::start() {
+void InternalTimer::start() {
 #if NODEMANAGER_TIME == ON
 	// target will be the current unix timestamp plus the requested
 	if (_mode == TIME_INTERVAL) _target = now() + _value;
@@ -64,12 +64,12 @@ void Timer::start() {
 }
 
 // stop the timer
-void Timer::stop() {
+void InternalTimer::stop() {
 	_is_running = false;
 }
 
 // update the timer and keep track of the elapsed time
-void Timer::update() {
+void InternalTimer::update() {
 #if NODEMANAGER_TIME == OFF && NODEMANAGER_SLEEP == ON
 	// if a sleeping node and time is not reliable take out from target the time slept in the previous cycle
 	if (_mode == TIME_INTERVAL && nodeManager.isSleepingNode()) _target -= nodeManager.getSleepSeconds()*1000UL;
@@ -77,7 +77,7 @@ void Timer::update() {
 }
 
 // return true if the time is over
-bool Timer::isOver() {
+bool InternalTimer::isOver() {
 	// time is never over if not configured or instructed to never report
 	if (_mode == DO_NOT_REPORT || _mode == NOT_CONFIGURED) return false;
 	// timer is always over when reporting immediately

--- a/nodemanager/InternalTimer.h
+++ b/nodemanager/InternalTimer.h
@@ -20,14 +20,14 @@
 #define Timer_h
 
 /******************************************
-Timer: helper class to keep track of the elapsed time
+InternalTimer: helper class to keep track of the elapsed time
 */
 
 class NodeManager;
 
-class Timer {
+class InternalTimer {
 public:
-	Timer();
+	InternalTimer();
 	void setMode(timer_mode mode);
 	timer_mode getMode();
 	void setValue(unsigned long value);

--- a/nodemanager/Node.cpp
+++ b/nodemanager/Node.cpp
@@ -555,10 +555,10 @@ void NodeManager::loop() {
 		_message.setType(type);
 		// send the message, multiple times if requested
 		for (int i = 0; i < _retries; i++) {
-			if (mGetPayloadType(_message) == P_INT16) debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%d\n"),_message.sensor,_message.type,_message.getInt());
-			if (mGetPayloadType(_message) == P_LONG32) debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%ld\n"),_message.sensor,_message.type,_message.getLong());
-			if (mGetPayloadType(_message) == P_FLOAT32) debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%d.%02d\n"),_message.sensor,_message.type,(unsigned int)_message.getFloat(), (unsigned int)(_message.getFloat()*100)%100);
-			if (mGetPayloadType(_message) == P_STRING) debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%s\n"),_message.sensor,_message.type,_message.getString());
+			if (mGetPayloadType(_message) == P_INT16) { debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%d\n"),_message.sensor,_message.type,_message.getInt()); }
+			if (mGetPayloadType(_message) == P_LONG32) { debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%ld\n"),_message.sensor,_message.type,_message.getLong()); }
+			if (mGetPayloadType(_message) == P_FLOAT32) { debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%d.%02d\n"),_message.sensor,_message.type,(unsigned int)_message.getFloat(), (unsigned int)(_message.getFloat()*100)%100); }
+			if (mGetPayloadType(_message) == P_STRING) { debug_verbose(PSTR(LOG_MSG "SEND(%d) t=%d p=%s\n"),_message.sensor,_message.type,_message.getString()); }
 			send(_message, _ack);
 			// if configured, sleep between each send
 			sleepBetweenSend();

--- a/nodemanager/Node.cpp
+++ b/nodemanager/Node.cpp
@@ -156,7 +156,7 @@ void NodeManager::before() {
 	// print out MySensors' library capabilities
 	debug(PSTR(LOG_INIT "LIB VER=" MYSENSORS_LIBRARY_VERSION " CP=" MY_CAPABILITIES " \n"));
 	// setup the reboot pin if needed
-	if (_reboot_pin > -1) {
+	if (_reboot_pin > 0) {
 		debug(PSTR(LOG_INIT "RBT p=%d\n"),_reboot_pin);
 		pinMode(_reboot_pin, OUTPUT);
 		digitalWrite(_reboot_pin, HIGH);
@@ -344,7 +344,7 @@ void NodeManager::loop() {
 	// reboot the board
 	void NodeManager::reboot() {
 		debug(PSTR(LOG_POWER "RBT\n"));
-		if (_reboot_pin > -1) {
+		if (_reboot_pin > 0) {
 			// reboot the board through the reboot pin which is connected to RST by setting it to low
 			digitalWrite(_reboot_pin, LOW);
 		}
@@ -477,7 +477,7 @@ void NodeManager::loop() {
 	// setup the interrupt pins
 	void NodeManager::_setupInterrupts() {
 		// configure wakeup pin if needed
-		if (_sleep_interrupt_pin > -1) {
+		if (_sleep_interrupt_pin > 0) {
 			// set the interrupt when the pin is connected to ground
 			setInterrupt(_sleep_interrupt_pin,FALLING,HIGH);
 		}
@@ -738,7 +738,7 @@ void NodeManager::setAnalogReference(uint8_t value, uint8_t pin) {
 	// wait a bit 
 	wait(200);
 	// perform some reading before actually reading the value
-	if (pin > -1) {
+	if (pin > 0) {
 		for (int i = 0; i < 5; i++) {
 			analogRead(pin);
 			wait(50);

--- a/nodemanager/Node.cpp
+++ b/nodemanager/Node.cpp
@@ -29,16 +29,6 @@ NodeManager::NodeManager(uint8_t sensor_count) {
 	_message = MyMessage();
 	// allocate block for all the sensors if sensor_count is provided
 	if (sensor_count > 0) sensors.allocateBlocks(sensor_count);
-	// setup serial port baud rate
-#ifndef CHIP_NRF5
-	MY_SERIALDEVICE.begin(MY_BAUD_RATE);
-#endif
-	// print out the version
-	debug(PSTR(LOG_INIT "VER=" VERSION "\n"));
-	// print out sketch name
-	debug(PSTR(LOG_INIT "INO=" SKETCH_NAME " v" SKETCH_VERSION "\n"));
-	// print out MySensors' library capabilities
-	debug(PSTR(LOG_INIT "LIB VER=" MYSENSORS_LIBRARY_VERSION " CP=" MY_CAPABILITIES " \n"));
 }
 
 #if NODEMANAGER_INTERRUPTS == ON
@@ -159,6 +149,12 @@ void NodeManager::registerTimer(Timer* timer) {
 
 // setup NodeManager
 void NodeManager::before() {
+	// print out the version
+	debug(PSTR(LOG_INIT "VER=" VERSION "\n"));
+	// print out sketch name
+	debug(PSTR(LOG_INIT "INO=" SKETCH_NAME " v" SKETCH_VERSION "\n"));
+	// print out MySensors' library capabilities
+	debug(PSTR(LOG_INIT "LIB VER=" MYSENSORS_LIBRARY_VERSION " CP=" MY_CAPABILITIES " \n"));
 	// setup the reboot pin if needed
 	if (_reboot_pin > -1) {
 		debug(PSTR(LOG_INIT "RBT p=%d\n"),_reboot_pin);

--- a/nodemanager/Node.cpp
+++ b/nodemanager/Node.cpp
@@ -142,7 +142,7 @@ void NodeManager::registerSensor(Sensor* sensor) {
 }
 
 // register a timer
-void NodeManager::registerTimer(Timer* timer) {
+void NodeManager::registerTimer(InternalTimer* timer) {
 	_timers.push(timer);
 }
 
@@ -247,8 +247,8 @@ void NodeManager::loop() {
 	powerOn();
 #endif
 	// update all the registered timers
-	for (List<Timer*>::iterator itr = _timers.begin(); itr != _timers.end(); ++itr) {
-		Timer* timer = *itr;
+	for (List<InternalTimer*>::iterator itr = _timers.begin(); itr != _timers.end(); ++itr) {
+		InternalTimer* timer = *itr;
 		timer->update();
 	}
 	// run loop for all the registered sensors

--- a/nodemanager/Node.h
+++ b/nodemanager/Node.h
@@ -181,7 +181,11 @@ private:
 	uint8_t _reboot_pin = -1;
 	void _present(uint8_t child_id, uint8_t type);
 	List<Timer*> _timers;
+#if defined(ARDUINO_ARCH_STM32F1)
+	uint8_t _analog_reference = -1;
+#else
 	uint8_t _analog_reference = DEFAULT;
+#endif
 #if NODEMANAGER_INTERRUPTS == ON
 	uint8_t _interrupt_1_mode = MODE_NOT_DEFINED;
 	uint8_t _interrupt_2_mode = MODE_NOT_DEFINED;

--- a/nodemanager/Node.h
+++ b/nodemanager/Node.h
@@ -84,7 +84,7 @@ public:
 	// sleep between send()
 	void sleepBetweenSend();
 	// set the analog reference to the given value and optionally perform some fake reading on the given pin
-	void setAnalogReference(uint8_t value, uint8_t pin = -1);
+	void setAnalogReference(uint8_t value, uint8_t pin = 0);
 #if NODEMANAGER_SLEEP == ON
 	// [3] set the duration (in seconds) of a sleep cycle
 	void setSleepSeconds(unsigned long value);
@@ -175,10 +175,10 @@ private:
 	void _sendMessage(uint8_t child_id, uint8_t type);
 	unsigned long _sleep_time = 0;
 	bool _sleep_or_wait = true;
-	uint8_t _sleep_interrupt_pin = -1;
+	uint8_t _sleep_interrupt_pin = 0;
 	unsigned int _sleep_between_send = 0;
 	unsigned long _report_interval_seconds = 10*60;
-	uint8_t _reboot_pin = -1;
+	uint8_t _reboot_pin = 0;
 	void _present(uint8_t child_id, uint8_t type);
 	List<InternalTimer*> _timers;
 #if defined(ARDUINO_ARCH_STM32F1)

--- a/nodemanager/Node.h
+++ b/nodemanager/Node.h
@@ -72,7 +72,7 @@ public:
 	// register a sensor
 	void registerSensor(Sensor* sensor);
 	// register a timer
-	void registerTimer(Timer* timer);
+	void registerTimer(InternalTimer* timer);
 	// return the next-available child id
 	uint8_t getAvailableChildId(uint8_t child_id = 0);
 	// list containing all the registered sensors
@@ -180,7 +180,7 @@ private:
 	unsigned long _report_interval_seconds = 10*60;
 	uint8_t _reboot_pin = -1;
 	void _present(uint8_t child_id, uint8_t type);
-	List<Timer*> _timers;
+	List<InternalTimer*> _timers;
 #if defined(ARDUINO_ARCH_STM32F1)
 	uint8_t _analog_reference = -1;
 #else

--- a/nodemanager/Sensor.cpp
+++ b/nodemanager/Sensor.cpp
@@ -292,7 +292,7 @@ void Sensor::setReceiveHook(void (*function)(Sensor* sensor, MyMessage* message)
 // virtual functions
 void Sensor::onSetup(){
 }
-void Sensor::onLoop(Child* child){}
+void Sensor::onLoop(Child* /*child*/){}
 
 // by default when a child receive a REQ message and the type matches the type of the request, executes its onLoop function
 void Sensor::onReceive(MyMessage* message){
@@ -302,7 +302,7 @@ void Sensor::onReceive(MyMessage* message){
 }
 void Sensor::onInterrupt(){}
 #if NODEMANAGER_OTA_CONFIGURATION == ON
-void Sensor::onOTAConfiguration(ConfigurationRequest* request) {}
+void Sensor::onOTAConfiguration(ConfigurationRequest* /*request*/) {}
 #endif
 
 // evaluate the timer and return true if can be considered over

--- a/nodemanager/Sensor.cpp
+++ b/nodemanager/Sensor.cpp
@@ -29,8 +29,8 @@ Sensor::Sensor() {
 Sensor::Sensor(int8_t pin) {
 	_pin = pin;
 	// initialize the timers
-	_report_timer = new Timer();
-	_measure_timer = new Timer();
+	_report_timer = new InternalTimer();
+	_measure_timer = new InternalTimer();
 	// register the sensor with the node
 	nodeManager.registerSensor(this);
 }
@@ -306,7 +306,7 @@ void Sensor::onOTAConfiguration(ConfigurationRequest* request) {}
 #endif
 
 // evaluate the timer and return true if can be considered over
-bool Sensor::_evaluateTimer(Timer* timer) {
+bool Sensor::_evaluateTimer(InternalTimer* timer) {
 	// timer is over
 	if (timer->isOver()) return true;
 	if (_first_run) {

--- a/nodemanager/Sensor.h
+++ b/nodemanager/Sensor.h
@@ -24,7 +24,7 @@ Sensor: provide functionalities common to all the sensors
 */
 
 #include "Node.h"
-#include "Timer.h"
+#include "InternalTimer.h"
 #include "Child.h"
 #if NODEMANAGER_POWER_MANAGER == ON
 #include "PowerManager.h"
@@ -124,9 +124,9 @@ protected:
 	unsigned int _samples = 1;
 	unsigned long _samples_interval = 0;
 	bool _first_run = true;
-	Timer* _report_timer;
-	Timer* _measure_timer;
-	bool _evaluateTimer(Timer* timer);
+	InternalTimer* _report_timer;
+	InternalTimer* _measure_timer;
+	bool _evaluateTimer(InternalTimer* timer);
 #if NODEMANAGER_INTERRUPTS == ON
 	int8_t _interrupt_pin = -1;
 	uint8_t _interrupt_mode = MODE_NOT_DEFINED;

--- a/sensors/SensorAnalogInput.h
+++ b/sensors/SensorAnalogInput.h
@@ -29,7 +29,11 @@ protected:
 	bool _reverse = false;
 	bool _output_percentage = true;
 	int _range_min = 0;
+#if defined(ARDUINO_ARCH_STM32F1)
+	int _range_max = 4096;
+#else
 	int _range_max = 1024;
+#endif
 	
 public:
 	SensorAnalogInput(int8_t pin, uint8_t child_id = 0): Sensor(pin) {

--- a/sensors/SensorDigitalOutput.h
+++ b/sensors/SensorDigitalOutput.h
@@ -31,7 +31,7 @@ protected:
 	int _wait_after_set = 0;
 	int _pulse_width = 0;
 	bool _invert_value_to_write = false;
-	Timer* _safeguard_timer = new Timer();
+	InternalTimer* _safeguard_timer = new InternalTimer();
 	
 public:
 	SensorDigitalOutput(int8_t pin, uint8_t child_id = 0): Sensor(pin) {

